pkg/odo/cli/component/delete.go

```diff
@@ -48,6 +48,9 @@ func (do *DeleteOptions) Complete(name string, cmd *cobra.Command, args []string
 
 // Validate validates the list parameters
 func (do *DeleteOptions) Validate() (err error) {
+	if do.Context.Project == "" || do.Application == "" {
+		return odoutil.ThrowContextError()
+	}
```

Comment:
This is Ctrl+C and Ctrl+V three times, could we throw this into its own function?

Reply:
Even if I write a function, the function will have to throw the error as suggested in https://github.com/openshift/odo/pull/1779#discussion_r291144081. Thus the `if else` will still be required.